pkg/daemon/gpsd.go

```diff
@@ -236,6 +241,9 @@ retry:
 						}
 						break
 					}
+					if g.Stopped() {
```

Comment:
Which for loop is this intended to break from?

Reply:
this loop 
for {
					//UbloxPollInit only initializes if not running
					ublx.UbloxPollInit()
					output := ublx.UbloxPollPull()

Reply:
fixed

Reply:
Can you move it out of the entire inner for loop entirely?  Maybe just after it instead?  The inner loop should be nearly instantaneous, so it seems a waste performance-wise to be grabbing and releasing the mutex so many times with the call to Stopped().